File: template/quicksort.go
package template

func quickSort(arr []int) []int {
	if len(arr) <= 1 {
		return arr
	}

	pivot := arr[len(arr)-1]
	var left, right []int

	for i := 0; i < len(arr)-1; i++ {
		if arr[i] < pivot {
			left = append(left, arr[i])
		} else {
			right = append(right, arr[i])
		}
	}
	left = quickSort(left)
	right = quickSort(right)

	return append(append(left, pivot), right...)
}

File: template/quicksort.py
from typing import List


def quickSort(arr: List[int]) -> List[int]:
    if len(arr) <= 1:
        return arr

    pivot = arr[len(arr) >> 1]
    left = [x for x in arr if x < pivot]
    middle = [x for x in arr if x == pivot]
    right = [x for x in arr if x > pivot]

    # for x in arr:
    #     if x < pivot:
    #         left.append(x)
    #     elif x == pivot:
    #         middle.append(x)
    #     else:
    #         right.append(x)

    return quickSort(left) + middle + quickSort(right)